board/safety/safety_subaru_legacy.h

```diff
@@ -0,0 +1,180 @@
+const int SUBARU_L_MAX_STEER = 2047; // 1s
+// real time torque limit to prevent controls spamming
+// the real time limit is 1500/sec
+const int SUBARU_L_MAX_RT_DELTA = 940;          // max delta torque allowed for real time checks
+const uint32_t SUBARU_L_RT_INTERVAL = 250000;    // 250ms between real time checks
+const int SUBARU_L_MAX_RATE_UP = 50;
+const int SUBARU_L_MAX_RATE_DOWN = 70;
+const int SUBARU_L_DRIVER_TORQUE_ALLOWANCE = 75;
+const int SUBARU_L_DRIVER_TORQUE_FACTOR = 10;
+const int SUBARU_L_STANDSTILL_THRSLD = 20;  // about 1kph
+
+
+const CanMsg SUBARU_L_TX_MSGS[] = {{0x161, 0, 8}, {0x164, 0, 8}};
+const int SUBARU_L_TX_MSGS_LEN = sizeof(SUBARU_L_TX_MSGS) / sizeof(SUBARU_L_TX_MSGS[0]);
+
+// TODO: do checksum and counter checks after adding the signals to the outback dbc file
+AddrCheckStruct subaru_l_rx_checks[] = {
+  {.msg = {{0x140, 0, 8, .expected_timestep = 10000U}}},
+  {.msg = {{0x371, 0, 8, .expected_timestep = 20000U}}},
+  {.msg = {{0x144, 0, 8, .expected_timestep = 50000U}}},
+};
+const int SUBARU_L_RX_CHECK_LEN = sizeof(subaru_l_rx_checks) / sizeof(subaru_l_rx_checks[0]);
+
+// TODO add legacy checksum check
+
+const uint16_t SUBARU_L_PARAM_FLIP_DRIVER_TORQUE = 1;
+bool subaru_l_flip_driver_torque = false;
+
+static int subaru_legacy_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
+
+  bool valid = addr_safety_check(to_push, subaru_l_rx_checks, SUBARU_L_RX_CHECK_LEN,
+                            NULL, NULL, NULL);
+
+  if (valid && (GET_BUS(to_push) == 0)) {
+    int addr = GET_ADDR(to_push);
+    if (addr == 0x371) {
+      int torque_driver_new;
+      torque_driver_new = (GET_BYTE(to_push, 3) >> 5) + (GET_BYTE(to_push, 4) << 3);
+      torque_driver_new = to_signed(torque_driver_new, 11);
+      if (subaru_l_flip_driver_torque) {
+        torque_driver_new = -1 * torque_driver_new;
+      }
+      update_sample(&torque_driver, torque_driver_new);
+    }
+
+    // enter controls on rising edge of ACC, exit controls on ACC off
+    if (addr == 0x144) {
+      int cruise_engaged = ((GET_BYTES_48(to_push) >> 17) & 1);
+      if (cruise_engaged && !cruise_engaged_prev) {
+        controls_allowed = 1;
+      }
+      if (!cruise_engaged) {
+        controls_allowed = 0;
+      }
+      cruise_engaged_prev = cruise_engaged;
+    }
+
+    // sample wheel speed, averaging opposite corners
+    if (addr == 0xD4) {
+      int subaru_speed = (GET_BYTES_04(to_push) >> 16) & 0xFFFF;  // FR
+      subaru_speed += GET_BYTES_48(to_push) & 0xFFFF;  // RL
+      subaru_speed /= 2;
+      vehicle_moving = subaru_speed > SUBARU_L_STANDSTILL_THRSLD;
+    }
+
+    if (addr == 0xD1) {
+      brake_pressed = ((GET_BYTES_04(to_push) >> 16) & 0xFF) > 0;
```

Comment:
use GET_BYTE instead of shifting and anding

Reply:
done